iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnBuffers.cpp

```diff
@@ -305,20 +305,29 @@ struct DotGeneralOpConversion
 //===----------------------------------------------------------------------===//
 
 namespace {
-/// Converts mhlo.convolution operation to linalg.conv op.
-struct ConvOpConversion
-    : public ConvertToLinalgBufferOp<ConvOpConversion, mhlo::ConvOp> {
-  using ConvertToLinalgBufferOp<ConvOpConversion,
+
+bool isDepthwiseConv(mhlo::ConvOp op) {
+  auto shape = op.rhs().getType().cast<ShapedType>().getShape();
+  auto numGroups =
+      shape[op.dimension_numbers().kernel_input_feature_dimension().getInt()];
+  return op.feature_group_count() > 1u && op.feature_group_count() == numGroups;
+}
+
+/// Converts depthwise mhlo.convolution operation to linalg.generic op.
+struct DepthwiseConvOpConversion
+    : public ConvertToLinalgBufferOp<DepthwiseConvOpConversion, mhlo::ConvOp> {
+  using ConvertToLinalgBufferOp<DepthwiseConvOpConversion,
                                 mhlo::ConvOp>::ConvertToLinalgBufferOp;
   LogicalResult apply(mhlo::ConvOp op, ArrayRef<Value> inputBuffers,
                       ArrayRef<Value> resultBuffers,
                       ConversionPatternRewriter &rewriter) const;
 };
 }  // namespace
 
-LogicalResult ConvOpConversion::apply(
+LogicalResult DepthwiseConvOpConversion::apply(
```

Comment:
This part actually is not changed that much. I only removed the needed variables for conv lowering, and remove the if-else branch (which check if this is a depthwise conv op). So it looks scaring...

Reply:
Enabling " Hide whitespace changes" in ⚙️  makes it clear.